core-addon/Core.js

```diff
@@ -498,8 +501,9 @@ module.exports = class Core {
    */
   async _fetchAvailableStudies() {
     try {
-      const request = await fetch(this._userArguments.availableStudiesURI);
-      return (await request.json()).data;
+      const result = await browser.firefoxPrivilegedApi.getRemoteSettings();
+      const studies = result[0].data;
```

Comment:
Do we need some defensive programming here? Is `result` guaranteed to be a valid array that has at least an item with the `data` property?

Reply:
Yes, this is definitely a "WIP" bit :) I'd like to unpack the datastructure in a more sane way. I may move this to the privileged API implementation, so it returns something that's more useful to us.

For `about:ion` we only care about maintaining a single collection (not a set of collections in a record which is quite common), so we could do something much simpler here.

Reply:
We could consider something like having a separate record for each study, but I'm not sure that's helpful since we really want to be able to ensure that a given set of studies are interoperable, it just seems like more overhead and work to have to do.

It's essentially the difference between maintaining one single large JSON file vs. a bunch of small independent JSON files, the latter feels a little cleaner somehow but I think it'd just be overhead with no upside in our case.

Reply:
> It's essentially the difference between maintaining one single large JSON file vs. a bunch of small independent JSON files, the latter feels a little cleaner somehow but I think it'd just be overhead with no upside in our case.

I agree!

Reply:
I may have mentioned this elsewhere, but I figured this out and it was having an incorrect document that we've been testing against (it includes the "envelope" from RS which we probably don't want, but that's what you get when you pull directly from the server via `fetch()`)

We should have a JSON schema and check against it IMHO, both for development and also for production.

Reply:
As I was implementing this, I had to refresh my memory about how we use RS in `about:ion`:

1. we *are* using a set of records in the collection, one for each study
2. the `RemoteSettings(KEY).get()` API returns all records together as an `Array` in JS

I reviewed the production config and set up the dev server correctly, and the current changeset seems to be working, including updating when RS does.